### PR TITLE
canonical: Update to `v0.6`

### LINKED ISSUE #72: Port to canonical 0.6

### PR DESCRIPTION
**Pull request by vlopes11**
Resolves #72

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,22 +1,22 @@
 [package]
 name = "phoenix-core"
-version = "0.10.0"
+version = "0.11.0-rc.0"
```
Comment (CPerezz) [inline Cargo.toml:3]: The version change should be done in a sepparated commit and in the `release-0.11` branch. Not here. This is the procedure we've been following for all the crates.

File: Cargo.toml [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
+rand_core = "0.6"
 dusk-bytes = "0.1"
-dusk-bls12_381 = {version = "0.6", default-features = false}
-dusk-jubjub = {version = "0.8", default-features = false}
-dusk-poseidon = {version = "0.20", default-features = false}
-dusk-pki = {version = "0.6", default-features = false}
-canonical = {version = "0.5", optional = true}
-canonical_derive = {version = "0.5", optional = true}
+dusk-bls12_381 = {version="0.8.0-rc.0", default-features=false}
+dusk-jubjub = {version="0.10.0-rc.0", default-features=false}
```
Comment (CPerezz) [inline Cargo.toml:11]: You can use 0.10.0 directly

File: Cargo.toml [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
-rand_core = "0.5"
+rand_core = "0.6"
 dusk-bytes = "0.1"
-dusk-bls12_381 = {version = "0.6", default-features = false}
-dusk-jubjub = {version = "0.8", default-features = false}
-dusk-poseidon = {version = "0.20", default-features = false}
-dusk-pki = {version = "0.6", default-features = false}
-canonical = {version = "0.5", optional = true}
-canonical_derive = {version = "0.5", optional = true}
+dusk-bls12_381 = {version="0.8.0-rc.0", default-features=false}
```
Comment (CPerezz) [inline Cargo.toml:10]: You can use 0.8.0 directly

**Review by CPerezz — CHANGES_REQUESTED** [review]
To me, the commit messages should be edited before merging. As well as the CHANGELOG entries.

- Use always the keywords at the beginning (Add, Change, Fix, Remove, Update etc)

Also the version as said in the messages should never be bumped here. Always in the release branch via commit.

Also, I would edit the commit message to follow the guidelines mentioned above and start with the keywords mentioned.

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Update canonical to `v0.6` [#72](https://github.com/dusk-network/phoenix-core/issues/72)
```
Comment (ZER0) [inline CHANGELOG.md:12]: Use the form:
```
- Update `canonical from `v0.5` to `v0.6` [#72](https://github.com/dusk-network/phoenix-core/issues/72)
```

When we deal with deps updates / changes.

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Crossover: use all attributes as hash inputs [#69](https://github.com/dusk-network/phoenix-core/issues/69)
 - Message: use all attributes as hash inputs [#69](https://github.com/dusk-network/phoenix-core/issues/69)
```
Comment (ZER0) [inline CHANGELOG.md:14]: When those were added? I know this is not part of this PR, but the format is not consistent with the changelog and therefore it should be fixed. We can do it before the release, at this point. But keep that in mind for the future.

**Review by ZER0 — CHANGES_REQUESTED** [review]
The PR LGTM but the commits messages needs to be rephrase.
Specifically, it uses a format that currently is reserved to the workspace, for example see 79962e9dd573a13c914055363971cc127953b93e:

```crossover: use all attributes as hash inputs ```

It looks like the `crossover` is a member of a `phoenix-core` workspace. Plus, is not using one of the changelog verbs. This hould be rephrase as:

```Change `Crossover::hash` to use all inputs``` or ```Change `Crossover`'s hash to use all inputs` or something like that.
- crossover: use all attributes as hash inputs

File: CHANGELOG.md (hunk 5 above)
Comment (vlopes11) [reply CHANGELOG.md:14]: What is the exact issue with the format?

Comment (CPerezz) [reply CHANGELOG.md:11]: That should start wit the `Update`, `Add`, `Remove` etc.. words.

**Review by CPerezz — APPROVED** [review]
LGTM

**Pull request merged**